### DISCUSSION TITLE
[a11y] Qtable compatibility with RGAA (French WCAG)

### DISCUSSION THREAD

**Discussion by NPi2Loup** [quasarframework/quasar#16115] [Ideas / Proposals]
Hello ! 
We're working on a French public portal. 
An external agency has carried out an accessibility audit. Its similar to #16109 .
But all improvements are welcome, which is why I'm putting here those that concern QTable only, and which could be handled independently of other corrections.
These corrections are easier to make directly in QTable than in a project, and improve Quasar's overall accessibility.

# Table
The `<table>` tag must define a title. There are 4 possibilities:
- Add a `<caption>` tag.
- add a `title` attribute
- add an `aria-label` attribute
- add an `aria-labelledby` attribute.

At present, we don't know how to add an attribute to the `<table>` tag itself. 
The `aria-label` attribute is perhaps the simplest solution. And the `caption` may be more complex as it needs to be hidden from the standard user. 

# Th
The `<th>` tags must have a `scope=col` attribute.
Currently, we have to rewrite all headers with the `header-slot`.

Best regards,

**Comment by NPi2Loup** [comment]
I've just proposed a fix to these problems:
https://github.com/NPi2Loup/quasar/commit/5cec534c8df312ab6f918ca0a96de0de887da79d

Is a PR not related to an issue accepted?

**Reply by NPi2Loup** [reply]
Some fix after Dan review https://github.com/NPi2Loup/quasar/commit/e09c67595769e4b4162684da58e5792b80d8c978

**Comment by NPi2Loup** [comment]
After a few explanations from our expert, it seems that using `aria-label` on `<table>` is not right in our case.
Indeed, if the table has a title, `aria-labelledby` should be used.
Fortunately, `QTable` adds the title and the table, so we can correlate them.

I'm going to propose some code.
What's the "right" way to add a random identifier to quasar? (to put it on `.q-table__title` and add `aria-labelledby` on `<table>`)

**Reply by yusufkandemir** [reply]
Use `uid()` to generate a unique identifier, set it as the `id` of the title, and set it as the `aria-labelledby` of the table. See an example implementation in QFab [here](https://github.com/quasarframework/quasar/blob/95493d022c528e55a2718feb4da9f5c583c013bd/ui/src/components/fab/QFab.js#L52-L152).

**Reply by NPi2Loup** [reply]
I will, thank you

**Reply by NPi2Loup** [reply]
Done : https://github.com/NPi2Loup/quasar/commit/395501c4d7f0ce34f2cf8c6566961a914f86ca15

I set the titleUid into marginalsScope because we need it when we override header-slot.
(In fact we override it to set a correct hx tag for title. WCAG rule)

**Reply by NPi2Loup** [reply]
How do I propose this code to the Quasar team ?
I've read https://github.com/quasarframework/quasar/blob/dev/CONTRIBUTING.md#pull-request-guidelines
Do I just have to make a PR on dev ?
Is a PR related to a11y considered as an issue ? as a new feature ? I think it's somewhere in between.

**Comment by NPi2Loup** [comment]
WCAG references : 
https://www.w3.org/WAI/tutorials/tables/caption-summary/
https://www.w3.org/WAI/tutorials/tables/one-header/
